### PR TITLE
Fixing qa found issues

### PR DESCRIPTION
**Pull request by dvladi77**
https://domaintools.atlassian.net/browse/QARELENG-1208

### CODE REVIEW

File: nmsg/kafkaio.c [hunk 1]
```diff
@@ -783,7 +783,8 @@ kafka_write(kafka_ctx_t ctx, const uint8_t *key, size_t key_len, const uint8_t *
 {
 	int res;
 	if (ctx == NULL || ctx->consumer || ctx->state != kafka_state_ready) {
-		free((void*) buf);
```
Comment (cmikk) [inline nmsg/kafkaio.c:786]: This chunk of the fix is not necessary -- `free(NULL)` is a no-op, and the cast does not dereference the pointer.

(Also, the `(void *)` cast is redundant here, and can be removed).

Comment (dvladi77) [reply nmsg/kafkaio.c:786]: I guess you missed the line here. Do you mean that if (buf != NULL) not neccessary
Also free((void*) buf); removes this warning
```
nmsg/kafkaio.c:787:30: warning: passing argument 1 of 'free' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  787 |                         free(buf);
```
Do you want me to remove (void*) ?

Comment (cmikk) [reply nmsg/kafkaio.c:786]: Correct. A `NULL` pointer can be passed to `free()`, so a NULL check before free-ing is redundant.

As for the `const` qualifier, that's a deeper problem. If `kafka_write` may free its `buf` argument, then that argument should not be `const`.

**Review by cmikk — APPROVED** [review]
Please squash when merging

**Review by reedjc — APPROVED** [review]
I no longer see the kafkaio.c static analyzer issue.

**Review by shw700 — APPROVED** [review]

**Pull request merged**